DHIS2/dutyday/reporter.py

```diff
@@ -0,0 +1,323 @@
+                    result = run_action(hostdetails[server], "cloning")
+                    add_to_report(server, "cloning", result, item.get('description'))
+
+            if "custom" == item.get("type"):
+                for server in item.get("servers"):
+                    result = run_action(hostdetails[server], "custom", item.get("command"))
+                    add_to_report(server, "custom", result, item.get('description'))
+
+            if "catalinaerrors" == item.get("type"):
+                for server in item.get("servers"):
+                    result = run_action(hostdetails[server], "catalinaerrors", hostdetails[server].get("catalina_file"))
+                    add_to_report(server, "catalinaerrors", result, item.get('description'))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Arguments --file: config file, --check-config check config file")
+    
+    parser.add_argument('--file', type=str, required=True, help='Path to the config file.')
+    parser.add_argument('--check-config', action='store_true', help='Mode in to check the config file.')
+    parser.add_argument('--check-servers', action='store_true', help='Mode in to check the server connection. ')
+    parser.add_argument('--update', action='store_true', help='Update report and logger files.')
+    parser.add_argument('--mode', type=str, help='Report mode print/json/html', default="print")
+    
+    args = parser.parse_args()
+
+    with open(args.file, 'r') as file:
+        config = json.load(file)
+        if not args.check_config:
+            load_servers(config)
+        if args.check_config:
+            validate_config(config)
+        elif args.check_servers:
+            check_servers()
+        elif args.update:
+            update_scripts(config)
+        else:
+            run_logger(config)
+            if args.mode == "json":
+                print(json.dumps(report_details))
+            elif args.mode == "print":
+                for server in report_details.keys():
+                    print("\n\n\n------------------------------------------------")
+                    print("------------------------------------------------")
+                    print("\n"+server+"\n")
+                    for action_dict in report_details[server]:
+                        for action_key,details in action_dict.items():
+                            print("------------------------"+server+"-----------------------")
+                            print("------------------------"+action_key+"-----------------------")
+                            print(details.get("description","Empty description")+"\n")
+                            print(details.get("result","Empty result")+"\n")
+                            print("------------------------END-----------------------")
+                    print("------------------------------------------------")
+                    print("------------------------------------------------")
```

Comment:
Suggestion/nitpick:
Make The "------------------------------------------------" prints a function like:

```python
separator = "------------------------------------------------"


def print_separator(separator, text=None):
    if text:
        half_separator = separator[:len(separator)//2]
        print(f"{half_separator}{text}{half_separator}")
    else:
        print(separator)
```

This will help modifying the output format faster